tests/test_exceptions.py

```diff
@@ -95,7 +95,11 @@ def ignore_pytest_unraisable_warning(f):
 
 
 # TODO: find out why this fails on PyPy, https://foss.heptapod.net/pypy/pypy/-/issues/3583
-@pytest.mark.xfail(env.PYPY, reason="Failure on PyPy 3.8 (7.3.7)", strict=False)
+@pytest.mark.xfail(
+    env.PYPY and sys.version_info >= (3, 8),
```

Comment:
```suggestion
    env.PYPY and sys.implementation.version < (7, 3, 8),
```

Reply:
I _think_ I saw this not xpassing. But I could try it.

Reply:
Nope.

Reply:
Sorry for the noise. I wonder if we can get this to a more minimal reproducer